### DISCUSSION TITLE
Using default parameters, TautomerEnumerator removes defined stereo centers

### DISCUSSION THREAD

**Discussion by apahl** [rdkit/rdkit#6852] [Show and tell]
This is something I noticed today.

The TautomerEnumerator, when used with the default parameters, removes defined stereo centers, when they are involved in tautomeric isomers. An example:

**The original molecule:**

```Python
from rdkit.Chem import AllChem as Chem
from rdkit.Chem import Draw
from rdkit.Chem.Draw import IPythonConsole

from rdkit.Chem.MolStandardize.rdMolStandardize import TautomerEnumerator, CleanupParameters  # new

smi = "[H][C@@]12CCCC[C@]1(CCNC2=O)C1=CC=CC=C1"
mol = Chem.MolFromSmiles(smi)
mol
```
![image](https://github.com/rdkit/rdkit/assets/9406836/c65da598-59a7-45ec-a363-6e57f617aa80)

**Generate the canonical tautomer using default parameters:**

```
mol = Chem.MolFromSmiles(smi)
te = TautomerEnumerator()
mol = te.Canonicalize(mol)
mol
```
![image](https://github.com/rdkit/rdkit/assets/9406836/1566efea-f997-4a3c-a19f-614923ed9d06)

**Set the relevant parameters to retain stereo information:**

```Python
cup = CleanupParameters()
cup.tautomerReassignStereo = True
cup.tautomerRemoveBondStereo = False
cup.tautomerRemoveSp3Stereo = False
te = TautomerEnumerator(cup)

smi = "[H][C@@]12CCCC[C@]1(CCNC2=O)C1=CC=CC=C1"
mol = Chem.MolFromSmiles(smi)
mol = te.Canonicalize(mol)
mol
```
![image](https://github.com/rdkit/rdkit/assets/9406836/79b54713-c8ba-449d-a649-857d2b94b336)

The following parameters (with their default values) are available for `CleanupParameters`:
```Python
cup = CleanupParameters()

for e in dir(cup):
    if not e.startswith("_"):
        print(e, getattr(cup, e))

# -> acidbaseFile 
# -> doCanonical True
# -> fragmentFile 
# -> largestFragmentChooserCountHeavyAtomsOnly False
# -> largestFragmentChooserUseAtomCount True
# -> maxRestarts 200
# -> maxTautomers 1000
# -> maxTransforms 1000
# -> normalizationsFile 
# -> preferOrganic False
# -> tautomerReassignStereo True
# -> tautomerRemoveBondStereo True
# -> tautomerRemoveIsotopicHs True
# -> tautomerRemoveSp3Stereo True
# -> tautomerTransformsFile 
```

**Comment by DavidACosgrove** [comment]
I believe that’s correct behaviour. Whilst the physico-chemical relevance
is debatable, one of the possible tautomers of this system is an enol into
the ring fusion. When that is reversed, the H atom can attack from either
side of the double bond so the chirality at the bridgehead atom is lost.

**Reply by apahl** [reply]
Yes, I understand the rationale behind the behaviour. But I was still surprised that it was the default behaviour, since it returns a different molecule entity and, as you say, the physico-chemical relevance will be debatable in many cases.

**Reply by DavidACosgrove** [reply]
You might find this interesting if you’re wondering about whether racemisation will occur on a particular system: https://onlinelibrary.wiley.com/doi/full/10.1002/anie.201709163

**Reply by apahl** [reply]
Very interesting paper, thank you for sharing.
